### PR TITLE
use defer instead of os.Exit(m.Run())

### PR DESCRIPTION
**Pull request by pmenglund**
as of go1.15 `testing.M` returns the return code of `m.Run()`, so it is possible to use `defer`

## Related Issue or Design Document

https://github.com/golang/go/issues/34129

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [ ] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

## Further comments

n/a

### CODE REVIEW

**Comment by pmenglund** [comment]
this fails on something unrelated to the PR:
```
Error: ../../../go/pkg/mod/github.com/docker/cli@v24.0.1+incompatible/cli/compose/template/template.go:242:14: undefined: strings.Cut
```
do you want me to fix the issue in this PR or a separate one?

**Comment by alnr** [comment]
This is a good improvement. Would you mind rebasing to get rid of the updated Copyright headers?

**Comment by pmenglund** [comment]
> This is a good improvement. Would you mind rebasing to get rid of the updated Copyright headers?

Seems like you did this already.  Let me know if there is anything else I should do...

File: docker/change.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright © 2023 Ory Corp
+// Copyright © 2024 Ory Corp
```
Comment (alnr) [inline docker/change.go]: Basically, can you revert all of these changes?

**Review by alnr — APPROVED** [review]
Thank you!

**Pull request merged**